src/components/TapsList.js

```diff
@@ -0,0 +1,120 @@
+// @flow
+
+import type { Tap, TapMutator } from 'brewskey.js-api';
+import type DAOEntityStore from '../stores/DAOEntityStore';
+import type { InfiniteLoaderChildProps } from './InfiniteLoader';
+
+import * as React from 'react';
+import { inject, observer } from 'mobx-react';
+import { withNavigation } from 'react-navigation';
+import { StyleSheet } from 'react-native';
+import { ListItem } from 'react-native-elements';
+import SwipeableFlatList from '../common/SwipeableFlatList';
+import SwipeableActionButton from '../common/SwipeableFlatList/SwipeableActionButton';
+// imported from experimental react-native
+// eslint-disable-next-line
+import SwipeableQuickActions from 'SwipeableQuickActions';
+import InfiniteLoader from './InfiniteLoader';
+
+const styles = StyleSheet.create({
+  // eslint-disable-next-line react-native/no-color-literals
+  listItem: {
+    backgroundColor: 'white',
+  },
+});
+
+type Props = {|
+  tapStore: DAOEntityStore<Tap, TapMutator>,
+  // todo add better typing
+  navigation: Object,
+|};
+
+// todo add pullToRefresh
+@withNavigation
+@inject('tapStore')
+@observer
+class TapsList extends React.Component<Props> {
+  _swipeableFlatListRef: ?SwipeableFlatList;
+
+  _fetchNextData = async (): Promise<void> => {
```

Comment:
How does this know when you've fetched all the items?

Reply:
it doesn't, it calls fetchNextData callback every time we reach the end of list...basically like pool to refresh but from the bottom.